src/main/java/Parser.java

```diff
@@ -0,0 +1,446 @@
+package duke;
+
+import duke.DukeException;
+import duke.Command;
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.Task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.FormatStyle;
+import java.util.Optional;
+
+/**
+ * Parser deals with the full commands given by the user.
+ * Inputs given are deciphered and corresponding responses are outputted.
+ */
+
+public class Parser {
+	public static Command parse(String fullCommand) throws DukeException {
+		if(fullCommand.equals("list")) {
+			return list();
+		} else if (fullCommand.equals("bye")) {
+			return bye();
+		} else if (getWord(fullCommand).equals("done")) {
+			return done(fullCommand);
+		} else if (getWord(fullCommand).equals("todo")) {
+			return todo(fullCommand);
+		} else if (getWord(fullCommand).equals("deadline")) {
+			return deadline(fullCommand);
+		} else if (getWord(fullCommand).equals("event")) {
+			return event(fullCommand);
+		} else if (getWord(fullCommand).equals("delete")) {
+			return delete(fullCommand);
+		} else if(getWord(fullCommand).equals("save")) {
+			return save();
+		} else if (getWord(fullCommand).equals("find")) {
+			return find(fullCommand);
+		} else{
+			throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+		}
+	}
+
+	/**
+	 * Returns the first word in the string indicated by the first blank space.
+	 * If there are no blank space, the entire string is returned.
+	 *
+	 * @param string
```

Comment:
Perhaps add a description for the return value for Javadoc comments to make it more complete? I noticed this issue in several other places as well.